core/contracts/financial_templates/implementation/Liquidatable.sol

```diff
@@ -313,69 +315,68 @@ contract Liquidatable is PricelessPositionManager {
                 // NOTE: This should never be below zero since we prevent (sponsorDisputePct+disputerDisputePct) >= 0 in
                 // the constructor when these params are set
                 FixedPoint.Unsigned memory payToLiquidator = tokenRedemptionValue.sub(sponsorDisputeReward).sub(
                     disputerDisputeReward
                 );
-                require(collateralCurrency.transfer(msg.sender, payToLiquidator.rawValue));
+                withdrawalAmount = withdrawalAmount.add(payToLiquidator);
                 delete liquidation.liquidator;
-                withdrawalMade = payToLiquidator.rawValue;
             }
+
             // Free up space once all locked collateral is withdrawn
-            if (collateralCurrency.balanceOf(address(this)) == 0) {
+            if (
+                liquidation.disputer == address(0) &&
+                liquidation.sponsor == address(0) &&
+                liquidation.liquidator == address(0)
+            ) {
                 delete liquidations[sponsor][id];
             }
-            //In the case of a failed dispute only the liquidator can withdraw.
+            // In the case of a failed dispute only the liquidator can withdraw.
         } else if (liquidation.state == Status.DisputeFailed && msg.sender == liquidation.liquidator) {
             // Pay LIQUIDATOR: lockedCollateral + dispute bond
             FixedPoint.Unsigned memory payToLiquidator = liquidation.lockedCollateral.add(disputeBondAmount);
-            require(collateralCurrency.transfer(msg.sender, payToLiquidator.rawValue));
-            withdrawalMade = payToLiquidator.rawValue;
-            delete liquidation.liquidator;
+            withdrawalAmount = payToLiquidator;
             delete liquidations[sponsor][id];
-            //If the state is pre-dispute but time has passed liveness then the dispute failed and the liquidator can withdraw
+            // If the state is pre-dispute but time has passed liveness then the dispute failed and the liquidator can withdraw
         } else if (liquidation.state == Status.PreDispute && msg.sender == liquidation.liquidator) {
             // Pay LIQUIDATOR: lockedCollateral
-            require(collateralCurrency.transfer(msg.sender, liquidation.lockedCollateral.rawValue));
-            withdrawalMade = liquidation.lockedCollateral.rawValue;
-            delete liquidation.liquidator;
+            withdrawalAmount = liquidation.lockedCollateral;
             delete liquidations[sponsor][id];
         }
-        if (withdrawalMade > 0) {
+
+        if (withdrawalAmount.isGreaterThan(0)) {
+            collateralCurrency.safeTransfer(msg.sender, withdrawalAmount.rawValue);
+
             // TODO: add this amount to the event in the issue #875.
             emit LiquidationWithdrawn(msg.sender);
-            return withdrawalMade;
+            return withdrawalAmount;
         }
         require(false);
```

Comment:
as discussed IRL, try adding an `else{...}` around this so that you don't have to return `withdrawalAmount`

Also, try replacing with a `revert`?

Reply:
Done.